goagen/codegen/workspace.go

```diff
@@ -310,7 +310,7 @@ func PackagePath(path string) (string, error) {
 		if gp, err := filepath.Abs(gopath); err == nil {
 			gopath = gp
 		}
-		if strings.HasPrefix(absPath, gopath) {
+		if filepath.HasPrefix(absPath, gopath) {
```

Comment:
I don't think this does what you think. See the [implementation](https://golang.org/src/path/filepath/path_unix.go?s=606:643#L13)
Also the [comment](https://golang.org/pkg/path/filepath/#HasPrefix) of the method indicates it should not be used.